src/subprocess.rs

```diff
@@ -35,10 +47,11 @@ impl std::error::Error for ExitStatusError {}
 impl std::fmt::Display for ExitStatusError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         let _ = write!(f, "subprocess exited with status: ");
-        if let Some(status) = self.status {
-            writeln!(f, "{status}")
-        } else {
-            writeln!(f, "unknown")
+
+        match self {
+            Self::ExitCode(code) => writeln!(f, "{code}"),
+            Self::Signal(signal) => writeln!(f, "{signal}"),
```

Comment:
We should expose the new detail we have:

```suggestion
            Self::Signal(signal) => writeln!(f, "signal {signal}"),
```